src/DataStructures/Python/DataVector.cpp

```diff
@@ -2,23 +2,44 @@
 // See LICENSE.txt for details.
 
 #include <boost/python.hpp>
+#include <iostream>
+#include <memory>
 #include <sstream>
 #include <string>
 #include <utility>
 
 #include "DataStructures/DataVector.hpp"
 #include "PythonBindings/BoundChecks.hpp"
 #include "Utilities/GetOutput.hpp"
-
 namespace bp = boost::python;
 
+// 3 "make" functions for 3 different constructors
+std::shared_ptr<DataVector> make_datavector_from_list(bp::list l) {
+  DataVector t = DataVector(bp::len(l));
+  for (int i = 0; i < bp::len(l); i++) {
+    t[i] = bp::extract<double>(l[i]);
+  }
+  return std::make_shared<DataVector>(std::move(t));
+}
+std::shared_ptr<DataVector> make_datavector_from_size(int size) {
+  return std::make_shared<DataVector>(static_cast<size_t>(size));
+}
+std::shared_ptr<DataVector> make_datavector_from_size_value(int size,
+                                                            double val) {
+  return std::make_shared<DataVector>(static_cast<size_t>(size), val);
+}
+
 namespace py_bindings {
 void bind_datavector() {
   // Wrapper for basic DataVector operations
-  bp::class_<DataVector>("DataVector", bp::init<size_t>())
-      .def(bp::init<size_t, double>())
+  bp::class_<DataVector>("DataVector", bp::no_init)
+      // Each "make" function gets its own constructor
+      .def("__init__", bp::make_constructor(&make_datavector_from_list))
+      .def("__init__", bp::make_constructor(&make_datavector_from_size))
+      .def("__init__", bp::make_constructor(&make_datavector_from_size_value))
```

Comment:
Do you need the latter two `make_constructor`s? I think these can remain `.def(bp::init<...>())`.

Reply:
@nilsleiffischer  `PyArrayObject` doesn't seem to be in `boost::python`, is it possibly in a later version then we are using?

Reply:
Oh right, it‘s in numpy: https://docs.scipy.org/doc/numpy-1.13.0/reference/c-api.types-and-structures.html
We can defer this if it’s too much effort to get working.

Reply:
New version of Boost have Boost.numpy, but that's too new for us to be able to use, unfortunately. @osheamonn might have some insight here since he's worked with Numpy arrays in a related context before.